### PR TITLE
 Flickable web pages

### PR DESCRIPTION
**Pull request by vladest**
https://highfidelity.fogbugz.com/f/cases/6610/Add-Momentum-to-Scrolling

For preview only:

Test plan:
- Open Tablet
- Goto Market/Info card on Address bar/Create menu lists, properties etc
- try to scroll web page. Make sure its flickable

Known issues:

HTML tables scrolling possible only by clicking on scroller

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2637/HighFidelity-Beta-PR11221-43d3544a0d6fadabd387dfe8085e8ab5ff26ae45.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2637/HighFidelity-Beta-PR11221-43d3544a0d6fadabd387dfe8085e8ab5ff26ae45.exe

**Comment by anshuman64** [comment]
@vladest This is working very well on the Marketplace page! 

Found another issue: Scrollbar in Marketplace is now missing. 

Great work, keep going!

**Comment by vladest** [comment]
@anshuman64 do you need scrollbar or scroll indicator?

**Comment by anshuman64** [comment]
@vladest Yes, we would like to keep those as original for now. We might want to remove them later though. Is it a lot of work to implement?

**Comment by vladest** [comment]
@anshuman64 in original, scroll bar was draw by web engine. now we need to use or scroll bar or scroll indicator from QQC. So, I assume that it should be scroll bar, not indicator, correct?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2782/HighFidelity-Beta-PR11221-36879f83eb6df2a0337a6d45f0f0e5ef8d3334a7.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2790/HighFidelity-Beta-PR11221-36879f83eb6df2a0337a6d45f0f0e5ef8d3334a7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2790/HighFidelity-Beta-PR11221-36879f83eb6df2a0337a6d45f0f0e5ef8d3334a7.exe

**Comment by anshuman64** [comment]
@vladest The flicking does not work anymore :(

**Comment by vladest** [comment]
@anshuman64 works for me. how did you tested it?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2835/HighFidelity-Beta-PR11221-e38075f7bad8e708a06b4d778411ac6d5c639240.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2835/HighFidelity-Beta-PR11221-e38075f7bad8e708a06b4d778411ac6d5c639240.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2902/HighFidelity-Beta-PR11221-88b045e3a97b71e8703807498b246898fafc88cb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2902/HighFidelity-Beta-PR11221-88b045e3a97b71e8703807498b246898fafc88cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2949/HighFidelity-Beta-PR11221-a80d9a22df59ad1692713f3214f3c0f90811967f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2949/HighFidelity-Beta-PR11221-a80d9a22df59ad1692713f3214f3c0f90811967f.exe

**Comment by anshuman64** [comment]
@vladest Two issues:

Create > List still dynamically resizes:
1. Go to your sandbox
2. Create 20 cubes (to populate your entity list beyond one page)
3. Exit Create mode and close the tablet. 
4. Reopen the tablet and go to Create > List.
Notice that the whole page has resized. It should stay the same size and only the table should be scrollable.

Create > Grid has a scrollbar:
Reset your settings and go to Create > Grid. Notice that there is a scrollbar. If it's not there the first time, close and reopen the tablet and check again.

If these issues are hard to fix, maybe we should undo your last commit (to allow scrolling again), merge this PR, and create separate tickets for those issues?

**Comment by anshuman64** [comment]
Updated Test Plan:
Go through all the default apps in the tablet. Make sure all pages are "flickable" (ie. you swipe up once and it continues scrolling for a while). Make sure scrollbars appear where they should (and only when they should). 

Please note any other weird behavior (blank spaces, double scrollbars) as you see it.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2958/HighFidelity-Beta-PR11221-6faab9fb1c8360dec61d76d24dd43efe466a5c07.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2958/HighFidelity-Beta-PR11221-6faab9fb1c8360dec61d76d24dd43efe466a5c07.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Go through all the default apps in the tablet. Make sure all pages are "flickable" (ie. you swipe up once and it continues scrolling for a while). Make sure scrollbars appear where they should (and only when they should). **Pass**

Please note any other weird behavior (blank spaces, double scrollbars) as you see it. **None found**

**Notes: The only thing which acted differently with the swipefeature was the embedded "List" in Create, the scroll bar for looking along the list is sharp in movement, but I would assume this is normal behaviour as I cannot see swipe/gliding movement to be appealing with this specific scroll-able area.**

**Comment by anshuman64** [comment]
@YuppyPlays Awesome thanks. Yea, we know there are issues on the List page--we'll address them in future PR's.

File: interface/resources/qml/TabletBrowser.qml [hunk 1]
```diff
@@ -17,7 +17,7 @@ Item {
     height: 600
     property variant permissionsBar: {'securityOrigin':'none','feature':'none'}
     property alias url: webview.url
-    property WebEngineView webView: webview
+    //property WebEngineView webView: webview
```
Comment (druiz17) [inline interface/resources/qml/TabletBrowser.qml]: Please remove commented/unused code

File: interface/resources/qml/controls/TabletWebView.qml [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+//import QtQuick.Controls 1.4
+import QtWebEngine 1.5
+//import QtWebChannel 1.0
 import "../controls-uit" as HiFiControls
 import "../styles" as HifiStyles
 import "../styles-uit"
-import "../"
-import "."
+//import "../"
+//import "."
```
Comment (druiz17) [inline interface/resources/qml/controls/TabletWebView.qml]: same here. Please remove commented code.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3059/HighFidelity-Beta-PR11221-85a3771a98ff790717c2c6d5139e3f472b85b98c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3059/HighFidelity-Beta-PR11221-85a3771a98ff790717c2c6d5139e3f472b85b98c.exe

**Pull request merged**